environment.py

```diff
@@ -32,6 +32,9 @@ def __init__(self, config):
         if self._si is None:
             raise ValueError("Unable to connect to host")
         self._ingest = self._create_signalfx_ingest()
+        self._additional_dims = None
```

Comment:
Do we need a function to do this?  Your function is just copying the dictionary. Can we set self._additional_dims to config['dimensions'] if it exists?

Maybe...
 `self._additional_dims = config.get('dimensions', {})`